libs/api/prisma/migrations/20250904071505_make_all_guids_in_the_database_strings/migration.sql

```diff
@@ -0,0 +1,176 @@
+  - The primary key for the `events.tagged-events` table will be changed. If it partially fails, the table could be left without primary key constraint.
+  - The primary key for the `mail.log` table will be changed. If it partially fails, the table could be left without primary key constraint.
+  - The primary key for the `mail_templates` table will be changed. If it partially fails, the table could be left without primary key constraint.
+  - The primary key for the `periodic_jobs` table will be changed. If it partially fails, the table could be left without primary key constraint.
+  - The primary key for the `subscription_communication_flows` table will be changed. If it partially fails, the table could be left without primary key constraint.
+  - The primary key for the `subscriptions.intervals` table will be changed. If it partially fails, the table could be left without primary key constraint.
+  - The primary key for the `user-consents` table will be changed. If it partially fails, the table could be left without primary key constraint.
+  - The primary key for the `user_communication_flows` table will be changed. If it partially fails, the table could be left without primary key constraint.
+
+*/
+-- DropForeignKey
+ALTER TABLE "_BannerToPage" DROP CONSTRAINT "_BannerToPage_A_fkey";
+
+-- DropForeignKey
+ALTER TABLE "_CrowdfundingToMemberPlan" DROP CONSTRAINT "_CrowdfundingToMemberPlan_A_fkey";
+
+-- DropForeignKey
+ALTER TABLE "_PaymentMethodToSubscriptionFlow" DROP CONSTRAINT "_PaymentMethodToSubscriptionFlow_B_fkey";
+
+-- DropForeignKey
+ALTER TABLE "banner_actions" DROP CONSTRAINT "banner_actions_bannerId_fkey";
+
+-- DropForeignKey
+ALTER TABLE "crowdfunding_goals" DROP CONSTRAINT "crowdfunding_goals_crowdfundingId_fkey";
+
+-- DropForeignKey
+ALTER TABLE "events.tagged-events" DROP CONSTRAINT "events.tagged-events_eventId_fkey";
+
+-- DropForeignKey
+ALTER TABLE "mail.log" DROP CONSTRAINT "mail.log_mailTemplateId_fkey";
+
+-- DropForeignKey
+ALTER TABLE "subscriptions.intervals" DROP CONSTRAINT "subscriptions.intervals_mailTemplateId_fkey";
+
+-- DropForeignKey
+ALTER TABLE "subscriptions.intervals" DROP CONSTRAINT "subscriptions.intervals_subscriptionFlowId_fkey";
+
+-- DropForeignKey
+ALTER TABLE "user-consents" DROP CONSTRAINT "user-consents_consentId_fkey";
+
+-- DropForeignKey
+ALTER TABLE "user_communication_flows" DROP CONSTRAINT "user_communication_flows_mailTemplateId_fkey";
+
+-- AlterTable
+ALTER TABLE "_BannerToPage" ALTER COLUMN "A" SET DATA TYPE TEXT;
+
+-- AlterTable
+ALTER TABLE "_CrowdfundingToMemberPlan" ALTER COLUMN "A" SET DATA TYPE TEXT;
+
+-- AlterTable
+ALTER TABLE "_PaymentMethodToSubscriptionFlow" ALTER COLUMN "B" SET DATA TYPE TEXT;
+
+-- AlterTable
+ALTER TABLE "banner_actions" DROP CONSTRAINT "banner_actions_pkey",
+ALTER COLUMN "id" SET DATA TYPE TEXT,
+ALTER COLUMN "bannerId" SET DATA TYPE TEXT,
+ADD CONSTRAINT "banner_actions_pkey" PRIMARY KEY ("id");
```

Comment:
UUIDs are pretty heavily normalized when being written to the database. Because the tables using a UUID were added when the database was already migrated to PostgreSQL, this probably doesn't affect us. But it might still be more cautious to use `citext` instead of `text` here. That way, comparisons and the unique constraint keep being case insensitive and the index is used regardless of the casing of the query string.

Reply:
Why not just rewrite existing non conform uuids with:
`UPDATE articles
SET id = gen_random_uuid()
WHERE NOT (
  uuid_column ~* '^[0-9a-f]{8}-[0-9a-f]{4}-[1-5][0-9a-f]{3}-[89ab][0-9a-f]{3}-[0-9a-f]{12}$'
);?
`